virtual-desktop/src/app/authentication-manager/login/login.component.ts

```diff
@@ -183,14 +186,46 @@ export class LoginComponent implements OnInit {
     }    
   }
 
+  cancelPasswordReset(): void {
+    this.loginMessage = "";
+    this.errorMessage = "";
+    this.expiredPassword = false;
+  }
+
+  loginMiddleman(): void {
+    if (this.expiredPassword) {
+      this.attemptPasswordReset();
+    } else {
+      this.attemptLogin();
+    }
+  }
+
+  attemptPasswordReset(): void {
+    this.authenticationService.performPasswordReset(this.username, this.password, this.newPassword).subscribe(
+      result => {
+        let jsonMessage = result.json();
+        this.expiredPassword = false;
+        this.loginMessage = jsonMessage.categories.zss.plugins['org.zowe.zlux.auth.zss'].response;
```

Comment:
do not hardcode this. store knowledge of which plugin is being asked for ahead of time, and look for the one you asked for, or since you only ask to reset 1 password just change the response body to just have the contents of that one plugin, without having the name of it.